### PR TITLE
Let `contentType` and `contentLength` return `Optional`s to get rid o…

### PR DESCRIPTION
**Pull request by dmfs**
…f `null` and `-1`. Implements #61

### CODE REVIEW

File: httpurlconnection-executor/src/main/java/org/dmfs/httpessentials/httpurlconnection/HttpUrlConnectionResponseEntity.java [hunk 1]
```diff
@@ -44,16 +47,18 @@ public HttpUrlConnectionResponseEntity(HttpURLConnection connection)
...
+        return mediaType == null ? Absent.<MediaType>absent() : new Present<MediaType>(new StringMediaType(mediaType));
     }
 
 
     @Override
-    public long contentLength() throws IOException
+    public Optional<Long> contentLength() throws IOException
     {
-        return mConnection.getContentLength();
+        long lenght = mConnection.getContentLength();
```
Comment (lemonboston) [inline httpurlconnection-executor/src/main/java/org/dmfs/httpessentials/httpurlconnection/HttpUrlConnectionResponseEntity.java]: I noticed that the javadoc of `.getContentLength()` advises to use `.getContentLengthLong()` instead.

File: http-client-basics/src/main/java/org/dmfs/httpessentials/responsehandlers/StringResponseHandler.java [hunk 2]
```diff
@@ -67,14 +81,12 @@ public String handleResponse(HttpResponse response) throws IOException, Protocol
...
+        Optional<MediaType> contentType = entity.contentType();
 
-        StringBuilder builder = new StringBuilder(
-                entity.contentLength() > 0 ? (int) entity.contentLength() : BUFFER_SIZE);
+        StringBuilder builder = new StringBuilder((int) (long) entity.contentLength().value((long) BUFFER_SIZE));
 
         // note: we already read the content in larger chunks so there should be no need for a BufferedReader
-        Reader reader = new InputStreamReader(entity.contentStream(),
-                contentType != null ? contentType.charset(mDefaultCharset) : mDefaultCharset);
+        Reader reader = new InputStreamReader(entity.contentStream(), contentType.value(mDefaultMediaType).charset(mDefaultCharset));
```
Comment (lemonboston) [inline http-client-basics/src/main/java/org/dmfs/httpessentials/responsehandlers/StringResponseHandler.java]: I guess `MediaType.charset()` could return `Optional` as well, right? In a separate task maybe?

Comment (dmfs) [inline http-client-basics/src/main/java/org/dmfs/httpessentials/responsehandlers/StringResponseHandler.java]: Right, that's a separate thing.

File: httpurlconnection-executor/src/main/java/org/dmfs/httpessentials/httpurlconnection/HttpUrlConnectionResponseEntity.java (hunk 1 above)
Comment (dmfs) [inline httpurlconnection-executor/src/main/java/org/dmfs/httpessentials/httpurlconnection/HttpUrlConnectionResponseEntity.java]: good catch!

**Pull request merged**